pystiche_papers/sanakoyeu_et_al_2018/_nst.py

```diff
@@ -210,3 +222,114 @@ def optim_loop(transformer: nn.Module) -> nn.Module:
             transformer_lr_scheduler.step(epoch)
 
     return transformer
+
+
+def training(
+    content_image_loader: DataLoader,
+    style_image_loader: DataLoader,
```

Comment:
Couldn't we also accept a `str` and use this as root to load the content and style dataset?

```suggestion
    content_image_loader: Union[DataLoader, str],
    style_image_loader: Union[DataLoader, str],
```

In case of the style dataset we could also introduce a `style: Optional[str] = None` parameter this is passed through the dataset. If this `is None` and `isinstance(style_image_loader, str)` we should raise an `ValueError` that `style` is needed for the style dataset. If on the other hand, `isinstance(style_image_loader, DataLoader)` we can simply ignore the `style` parameter or emit a warning that it has no effect.